### PR TITLE
enhancement: detection results to file

### PR DESCRIPTION
**Pull request by builderjer**
This writes the detection results to `~/.config/mycroft/i2c_platform` so that it can be read by plugins and other scripts.

### CODE REVIEW

**Comment by builderjer** [comment]
This enhancement was suggested by @JarbasAl so that other scripts can easily find out what sound card is installed.  I'm not sure if the added service should be run as a user instead though.

https://github.com/OpenVoiceOS/ovos-i2csound/issues/7

**Comment by builderjer** [comment]
The button is not working correctly with kernel 6.6 yet, so I eliminated them for now.

**Review by j1nx — APPROVED** [review]
LGTM

just as feedback. Perhaps also set the variable as environment variable so it can be used everywhere.

File: ovos-i2csound [hunk 1]
```diff
@@ -272,7 +290,12 @@ main() {
         amixer -c "aiyvoicebonnet" cset numid=47 on
         amixer -c "aiyvoicebonnet" cset numid=49 on
         amixer -c "aiyvoicebonnet" cset numid=27 on
+        i2c_device_name="AIYVOICEBONNET"
+    fi
 
+    # Write the detection results to a file only if something is detected
+    if [ "$i2c_device_name" ]; then
+        echo "$i2c_device_name" > /home/ovos/.config/mycroft/i2c_platform
```
Comment (NeonDaniel) [inline ovos-i2csound:298]: Does not check for directory existence. Could this use an envvar or something to be configurable? Alternatively, a path in `/opt` or `/etc` wouldn't depend on a particular username

**Pull request merged**